app/src/main/java/com/babylon/wallet/android/presentation/status/transaction/TransactionStatusDialogNav.kt

```diff
@@ -57,20 +45,8 @@ fun NavController.transactionStatusDialog(transactionEvent: AppEvent.Status.Tran
     // New requests will be handled from the view model itself
     if (currentBackStackEntry?.destination?.route?.startsWith(ROUTE) == true) return
 
-    val errorSerialized = if (transactionEvent is AppEvent.Status.Transaction.Fail) {
-        Json.encodeToString(transactionEvent.errorMessage)
-    } else {
-        null
-    }
-
-    val requestId = transactionEvent.requestId.ifBlank { error("Transaction id cannot be empty") }
-    navigate(
-        route = "$ROUTE/${transactionEvent.toType()}/$requestId" +
-            "?txId=${transactionEvent.transactionId}" +
-            "&isInternal=${transactionEvent.isInternal}" +
-            "&error=$errorSerialized" +
-            "&$ARG_BLOCK_UNTIL_COMPLETE=${transactionEvent.blockUntilComplete}"
-    ) {
+    val argument = Uri.encode(Serializer.kotlinxSerializationJson.encodeToString(transactionEvent.toTransactionStatusFields()))
```

Comment:
I would avoid using `Serializer.kotlinxSerializationJson`. This is related to gateway data parsing. Why is it needed here?

Reply:
> I would avoid using `Serializer.kotlinxSerializationJson`. This is related to gateway data parsing. Why is it needed here?

Good spot 👍 Will get rid of it.